internal/http/check.go

```diff
@@ -164,28 +182,29 @@ type Job struct {
 // Shutdown is needed to fulfil the check interface and in this case we are
 // shutting down the http server and waiting
 func (c *Check) Shutdown() error {
+	if c.shutingDown() {
+		return nil
+	}
+	c.inShutdown.Store(true)
```

Comment:
Now that you don't send a different response depending on the state of the server, I'd just remove the atomic.

Reply:
See 6f35590

Reply:
I mean `inShutdown` all together.